src/ReinforcementLearningCore/src/core/hooks.jl

```diff
@@ -104,32 +104,44 @@ Base.getindex(h::RewardsPerEpisode) = h.rewards
 Store the total reward of each episode in the field of `rewards`. If
 `is_display_on_exit` is set to `true`, a unicode plot will be shown at the [`PostExperimentStage`](@ref).
 """
-Base.@kwdef mutable struct TotalRewardPerEpisode <: AbstractHook
-    rewards::Vector{Float64} = Float64[]
-    reward::Float64 = 0.0
-    is_display_on_exit::Bool = true
+mutable struct TotalRewardPerEpisode{T} <: AbstractHook where {T<:Union{Bool,Nothing}}
```

Comment:
Is this a type stability change ?

Reply:
I have the same question for most changes below in fact. I guess the point is to use multiple dispatch to compile different functions (one that plots and one that doesn't)? Does this change significantly impact performance ?

Reply:
The plot and ProgressMeter functions seems to be type unstable, so if you don't need them, it's better if they aren't run. As a second thing, the plot / progress meter status was being checked at every hook / stop call, even though this is already known at type creation. So two reasons why it's useful; admittedly the Bool/Nothing in `hooks.jl` is a bit weird, but it follows the existing pattern in `stop.jl` with the progress meter